bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_06/D2_06_01.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enocean.internal.eep.D2_06;
+
+import static org.openhab.binding.enocean.internal.EnOceanBindingConstants.*;
+
+import java.util.function.Function;
+
+import org.openhab.binding.enocean.internal.eep.Base._VLDMessage;
+import org.openhab.binding.enocean.internal.messages.ERP1Message;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.CommonTriggerEvents;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ *
+ * @author Thomas Lauterbach - Initial contribution
+ */
+public class D2_06_01 extends _VLDMessage {
```

Comment:
Please add @NonNullByDefault

Reply:
Hi @lolodomo 

I added the annotation but I'm struggling with the annotation of the getCurrentStateFunc parameter of the convertToStateImpl method. If I annotate it with Nullable it will still give me an Exception due to a mismatched annotation. I think this might be due to the fact that State is annotated as NonNull but I can't figure out how to fix that... Any help would be highly appreciated, thank you!

Reply:
Maybe the problem is due to parent classes not being null annotated? 
Did you try to do annotate them too? 
If this is the problem, maybe we can release my comment as my intention was not to tell you to annotate many classes.

Reply:
This could be the case but it would go way beyond the scope of this PR to annotate large parts of the binding (no null annotations are used so far).

The correct way to do this for my single class (as the start of an iterative approach) would be as I understand it from the documentation to annotate the parameters with Nullable. This works fine for all parameter except for the Function<String, State> parameter with its generic types. What's special is that the State type is an openHAB core class that is properly annotated while the others are plain basic Java classes that are not annotated at all. I tried a lot of different things without really knowing what I'm doing and finally gave up yesterday night...

Reply:
Ok so please forget this in the context of this PR and push your updates.